flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/DynamicTableFactory.java

```diff
@@ -85,5 +88,23 @@
 
         /** Whether the table is temporary. */
         boolean isTemporary();
+
+        /**
+         * Shortcut for {@code getCatalogTable().getResolvedSchema().toPhysicalRowDataType()}.
+         *
+         * @see ResolvedSchema#toPhysicalRowDataType()
+         */
+        default FieldsDataType getPhysicalRowDataType() {
+            return getCatalogTable().getResolvedSchema().toPhysicalRowDataType();
+        }
+
+        /**
+         * Shortcut for {@code getCatalogTable().getResolvedSchema().getPrimaryKeyFields()}.
+         *
+         * @see ResolvedSchema#getPrimaryKeyFields()
+         */
+        default List<String> getPrimaryKeyFields() {
```

Comment:
as mentioned in the JIRA issue, a `int[]` would be more useful. in general, we don't handle this nicely at the moment. actually according to SQL a column must not have a unique name. positions might be more useful for connector developers.

Reply:
Gotcha, what about naming it `getPrimaryKeyIndexes` or `getPrimaryKeyFieldsIndexes`?